angular-workspace/projects/ni/nimble-angular/src/directives/theme-provider/tests/nimble-theme-provider.directive.spec.ts

```diff
@@ -11,4 +13,199 @@ describe('Nimble theme provider', () => {
+            fixture = TestBed.createComponent(TestHostComponent);
+            fixture.detectChanges();
+            directive = fixture.componentInstance.directive;
+            nativeElement = fixture.componentInstance.elementRef.nativeElement;
+        });
+
+        it('can be configured with property binding for theme', () => {
+            expect(directive.theme).toBe(Theme.dark);
+            expect(nativeElement.theme).toBe(Theme.dark);
+
+            fixture.componentInstance.theme = Theme.color;
+            fixture.detectChanges();
+
+            expect(directive.theme).toBe(Theme.color);
+            expect(nativeElement.theme).toBe(Theme.color);
+        });
+
+        it('can be configured with property binding for lang', () => {
+            expect(directive.lang).toBe('de-DE');
+            expect(nativeElement.lang).toBe('de-DE');
+
+            fixture.componentInstance.lang = '123';
+            fixture.detectChanges();
+
+            expect(directive.lang).toBe('123');
+            expect(nativeElement.lang).toBe('123');
+            expect(directive.validity.invalidLang).toBeTrue();
+            expect(directive.checkValidity()).toBeFalse();
+        });
+    });
+
+    describe('with attribute bound values', () => {
+        @Component({
+            template: `
+                <nimble-theme-provider #themeProvider
+                    [attr.theme]="theme"
+                    [attr.lang]="lang">
+                </nimble-theme-provider>
+            `
+        })
+        class TestHostComponent {
+            @ViewChild('themeProvider', { read: NimbleThemeProviderDirective }) public directive: NimbleThemeProviderDirective;
+            @ViewChild('themeProvider', { read: ElementRef }) public elementRef: ElementRef<ThemeProvider>;
+            public theme: Theme = Theme.dark;
+            public lang = 'de-DE';
+        }
+
+        let fixture: ComponentFixture<TestHostComponent>;
+        let directive: NimbleThemeProviderDirective;
+        let nativeElement: ThemeProvider;
+
+        beforeEach(() => {
+            TestBed.configureTestingModule({
+                declarations: [TestHostComponent],
+                imports: [NimbleThemeProviderModule]
+            });
+            fixture = TestBed.createComponent(TestHostComponent);
+            fixture.detectChanges();
+            directive = fixture.componentInstance.directive;
+            nativeElement = fixture.componentInstance.elementRef.nativeElement;
+        });
+
+        it('can be configured with attribute binding for theme', () => {
+            expect(directive.theme).toBe(Theme.dark);
+            expect(nativeElement.theme).toBe(Theme.dark);
+
+            fixture.componentInstance.theme = Theme.color;
+            fixture.detectChanges();
+
+            expect(directive.theme).toBe(Theme.color);
+            expect(nativeElement.theme).toBe(Theme.color);
+        });
+
+        it('can be configured with attribute binding for lang', () => {
```

Comment:
Same here. I think this test should just test the attribute binding with 'lang' being updated to a valid value.